server/src/database/entity/object.rs

```diff
@@ -87,6 +93,27 @@ pub enum Relation {
     Nar,
 }
 
+impl InsertExt for Insert<ActiveModel> {
+    fn on_conflict_do_update(self) -> Self {
+        self.on_conflict(
+            OnConflict::columns([Column::CacheId, Column::StorePathHash])
+                .update_columns([
+                    Column::NarId,
+                    Column::StorePath,
+                    Column::References,
+                    Column::System,
+                    Column::Deriver,
+                    Column::Sigs,
+                    Column::Ca,
+                    Column::CreatedAt,
+                    Column::LastAccessedAt,
+                    Column::CreatedBy,
```

Comment:
That’s a good point. I wanted to be able to overwrite existing objects including metadata (admittedly not very easy to do right now), and I assumed this scenario shouldn’t occur too often because clients should already filter out already-pushed paths via the `get-missing-paths` endpoint prior to pushing. Perhaps we want to make `on conflict do nothing` the default behavior and have a `force-overwrite` toggle?

But anyways, the push path is a pile of spaghetti and can have even more moving parts later on. Will need to finally give it some love it deserves.

Reply:
As you work to untangle the spaghetti, feel free to ping me on whatever PRs -- I'd love to take a look, for curiosity's sake!

(though I am subscribed to this repo, so I'll see it anyways -- I guess it's not really necessary unless you would specifically like another pair of eyes on something)